pulpcore/pulpcore/app/models/content.py

```diff
@@ -4,14 +4,64 @@
 import hashlib
 
 from django.core import validators
-from django.db import models
+from django.db import IntegrityError, models
+from django.forms.models import model_to_dict
+
 from itertools import chain
 
 from pulpcore.app.models import Model, MasterModel, Notes, GenericKeyValueRelation, storage, fields
 from pulpcore.exceptions import DigestValidationError, SizeValidationError
 
 
-class Artifact(Model):
+class BulkCreateManager(models.Manager):
+    """
+    A manager that provides a bulk_get_or_create()
+    """
+    def bulk_get_or_create(self, objs, batch_size=None):
+        """
+        Insert the list of objects into the database and get existing objects from the database.
+
+        Do *not* call save() on each of the instances, do not send any pre/post_save signals,
+        and do not set the primary key attribute if it is an autoincrement field (except if
+        features.can_return_ids_from_bulk_insert=True). Multi-table models are not supported.
+
+        If an IntegrityError is raised while performing a bulk insert, this method falls back to
+        inserting each instance individually. The already-existing instances are retrieved from
+        the database and returned with the other newly created instances.
+
+        Args:
+            objs (iterable of models.Model): an iterable of Django Model instances
+            batch_size (int): how many are created in a single query
+
+        Returns:
+            List of instances that were inserted into the database.
+        """
+        try:
+            return super().bulk_create(objs, batch_size=batch_size)
+        except IntegrityError:
+            objs = list(objs)
+            for i in range(len(objs)):
+                try:
+                    objs[i].save()
+                except IntegrityError:
+                    objs[i] = objs[i].objects.filter(objs[i].q())
```

Comment:
I didn't test this, but this looks like it should be a `get()`.

Reply:
Both work actually, but I did switch it to get().